src/document/mod.rs

```diff
@@ -84,113 +116,241 @@ impl DocumentEncryptOpts {
-
-    /// Update a document name to a new value or clear its value.
+    /// Returns the updated metadata of the document.
     ///
     /// # Arguments
-    /// - `id` - ID of the document to update
-    /// - `name` - New name for the document. Provide a Some to update to a new name and a None to clear the name field.
+    /// - `id` - Unique ID of the document to update
+    /// - `name` - New name for the document. Provide a `Some` to update to a new name or a `None` to clear the name field.
     ///
-    /// # Returns
-    /// `Result<DocumentMetadataResult>` Metadata about the document that was updated.
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// # let document_id: DocumentId = unimplemented!();
+    /// use std::convert::TryFrom;
+    /// let new_name = DocumentName::try_from("updated")?;
+    /// let document_meta = sdk.document_update_name(&document_id, Some(&new_name)).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
     async fn document_update_name(
         &self,
         id: &DocumentId,
         name: Option<&DocumentName>,
     ) -> Result<DocumentMetadataResult>;
 
-    /// Grant access to a document. Recipients of document access can be either users or groups.
+    /// Grants decryption access to a document for the provided users and/or groups.
+    ///
+    /// Returns lists of successful and failed grants.
     ///
     /// # Arguments
-    /// `document_id` - id of the document whose access is is being modified
-    /// `grant_list` - list of grants. Elements represent either a user or a group.
+    /// `document_id` - Unique ID of the document whose access is being modified.
+    /// `grant_list` - List of users and groups to grant access to.
     ///
-    /// # Returns
-    /// Outer result indicates that the request failed either on the client or that the server rejected
-    /// the whole request. If the outer result is `Ok` then each individual grant to a user/group
-    /// either succeeded or failed.
+    /// # Errors
+    /// This operation supports partial success. If the request succeeds, then the resulting
+    /// `DocumentAccessResult` will indicate which grants succeeded and which failed, and it
+    /// will provide an explanation for each failure.
+    ///
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// # let document_id: DocumentId = unimplemented!();
+    /// # let users: Vec<UserId> = vec![];
+    /// use std::convert::TryFrom;
+    /// use ironoxide::document::UserOrGroup;
+    /// // from a list of UserIds, `users`
+    /// let users_or_groups: Vec<UserOrGroup> = users.iter().map(|user| user.into()).collect();
```

Comment:
This is a great example as the [UserId] -> [UserOrGroup] conversion can be confusing. It also makes me want to make that easier, somehow...

Reply:
I agree that it can be confusing. Even worse is if you have a [UserId] and a [GroupId] and want to combine them into a [UserOrGroup]. I think it requires two mutable vecs and multiple lines. 
I would also love to make that process easier.